pkgs/tools/networking/ssldump/default.nix

```diff
@@ -30,9 +30,12 @@ stdenv.mkDerivation rec {
     openssl
   ];
 
-  prePatch = ''
+  # fdatasync function prototype is needed for darwin as it exists but is not
+  # declared in unistd.h
```

Comment:
I am not sure this is right. The only definition of `fdatasync` I can find in the SDK has this prototype:

```c
int fdatasync(struct proc *, struct fdatasync_args *, int *);
```

Are you sure this works correctly at runtime?

Reply:
Ah, I guess those are actually syscall prototypes. But I’m a little puzzled about this function that does not appear to be defined in any header, then.

Reply:
idk I saw this: https://www.postgresql.org/message-id/1673109.1610733352%40sss.pgh.pa.us and then just went for it.

Reply:
This was changed upstream: https://github.com/adulau/ssldump/commit/e5f2916b650252200c88374e8dc63b88db0613aa. Since they don’t seem to do releases on a regular basis and this package is already behind the latest tagged release, I suggest bumping to `1.8-unstable-…` instead.

https://www.postgresql.org/message-id/1673109.1610733352%40sss.pgh.pa.us suggests that this undocumented function may not actually function properly?

Reply:
> https://www.postgresql.org/message-id/1673109.1610733352%40sss.pgh.pa.us suggests that this undocumented function may not actually function properly?

I don't think that is what that thread said. in the end they stuck with it. if it is good enough for Postgres seems good enough for ssldump

Reply:
Fair enough. But this would become dead code after we bump the version anyway, so maybe we should just do that? There have been a bunch of fixes since the last tag. Or we could just fetch the commit patch, which will at least fail appropriately when it’s due for removal.